File: Delegate Pattern/DelegatePattern_Objc/DelegatePattern_Objc/VC/AViewController.m
//
//  ViewController.m
//  DelegatePattern_Objc
//
//  Created by leeyeon2 on 2021/07/25.
//

#import "AViewController.h"
#import "BViewController.h"
//2. 델리게이트 채택
@interface AViewController () <BViewControllerDelegate>
@property (weak, nonatomic) IBOutlet UILabel *messageFromBVCLabel;

@end

@implementation AViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (IBAction)showBVC:(id)sender {
    BViewController *bVC = [self.storyboard instantiateViewControllerWithIdentifier:@"BViewController"];
    //3. 위임자 설정!!
    //object.delegate = self
    bVC.delegate = self;
    bVC.modalPresentationStyle = UIModalPresentationFullScreen;
    [self presentViewController:bVC animated:YES completion:nil];
}

// 델리게이트 메소드 
-(void)sendMessage:(NSString*)message{
    _messageFromBVCLabel.text = message;
}


@end

File: Delegate Pattern/DelegatePattern_Swift/DelegatePattern_Swift/VC/AViewController.swift
//
//  ViewController.swift
//  DelegatePattern_Swift
//
//  Created by leeyeon2 on 2021/07/25.
//

import UIKit
//2. 델리게이트 채택
class AViewController: UIViewController, BViewControllerDelegate{

    @IBOutlet weak var messageFromBVCLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        print(String(describing: AViewController.self))
        print(String(describing: type(of: self)))
    }
    
    @IBAction func showBVC(_ sender: Any) {
        let storyboard = UIStoryboard(name: "Main", bundle: nil)
        let bVC: BViewController = storyboard.instantiateViewController(withIdentifier: "BViewController") as! BViewController
        bVC.modalPresentationStyle = .fullScreen
        self.present(bVC, animated: true, completion: nil)
        //3. 위임자 설정!!
        bVC.delegate = self
    }
    
    // 델리게이트 메소드
//    func sendMessage(message: String){
//        messageFromBVCLabel.text = message
//    }

    
    @IBAction func showXibVC(_ sender: Any) {
        let xibVC = XibViewController(nibName: "XibViewController", bundle: Bundle(for: XibViewController.self))
        self.present(xibVC, animated: true, completion: nil)
    }
    
}

